server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/SubAddressingTest.java

```diff
@@ -0,0 +1,220 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+
+public class SubAddressingTest {
+
+    private static final String SENDER1 = "sender1@localhost";
+    private static final String SENDER2 = "sender2@localhost";
+    private static final String RECIPIENT = "recipient@localhost";
+    private static final String TARGET = "targetfolder";
+    private static final String UNEXISTING_TARGET = "unexistingfolder";
+
+    private MailboxManager mailboxManager;
+    private SubAddressing testee;
+    private UsersRepository usersRepository;
+    private Username sender1Username;
+    private Username recipientUsername;
+    private MailboxSession recipientSession;
+    private MailboxId targetMailboxId;
+
+    @BeforeEach
+    void setUp() throws MailboxException, MessagingException, UsersRepositoryException {
+        mailboxManager = InMemoryIntegrationResources.defaultResources().getMailboxManager();
+        usersRepository = MemoryUsersRepository.withVirtualHosting(NO_DOMAIN_LIST);
+
+        recipientUsername = usersRepository.getUsername(new MailAddress(RECIPIENT));
+        sender1Username = usersRepository.getUsername(new MailAddress(SENDER1));
+
+        recipientSession = mailboxManager.createSystemSession(recipientUsername);
+        targetMailboxId = mailboxManager.createMailbox(
+            MailboxPath.forUser(recipientUsername, TARGET), recipientSession).get();
+
+        testee = new SubAddressing(usersRepository, mailboxManager);
+        testee.init(FakeMailetConfig.builder().build());
+    }
+
+    @Test
+    void shouldNotAddStorageDirectiveWhenTargetMailboxDoesNotExist() throws Exception {
+        Mail mail = mailBuilder(UNEXISTING_TARGET).sender(SENDER1).build();
+        testee.service(mail);
+
+        AttributeName recipient = AttributeName.of("DeliveryPaths_recipient@localhost");
+        assertThat(mail.attributes().map(this::unbox))
+            .doesNotContain(Pair.of(recipient, UNEXISTING_TARGET));
+    }
+
+    @Test
+    void shouldNotAddStorageDirectiveWhenNobodyHasRight() throws Exception {
+        givePostRightForKey(MailboxACL.ANYBODY_NEGATIVE_KEY);
+
+        Mail mail = mailBuilder(TARGET).sender(SENDER1).build();
+        testee.service(mail);
+
+        AttributeName recipient = AttributeName.of("DeliveryPaths_recipient@localhost");
+        assertThat(mail.attributes().map(this::unbox))
+            .doesNotContain(Pair.of(recipient, TARGET));
+    }
+
+
+    @Test
+    void shouldAddStorageDirectiveWhenAnybodyHasRight() throws Exception {
+        givePostRightForKey(MailboxACL.ANYBODY_KEY);
+
+        Mail mail = mailBuilder(TARGET).sender(SENDER1).build();
+        testee.service(mail);
+
+        AttributeName recipient = AttributeName.of("DeliveryPaths_recipient@localhost");
+        assertThat(mail.attributes().map(this::unbox))
+            .containsOnly(Pair.of(recipient, TARGET));
+    }
+
+    @Disabled
```

Comment:
Why is this test disabled? Usually we leave a reason with it like, so that if one day we want to fix it, we can remember quickly what was wrong in the first place

```
@Disabled("[]REASON]")
```

Reply:
final fixups were required as the test was still red, but it should be good now ;)